ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Background.hs

```diff
@@ -244,17 +247,26 @@ copyAndSnapshotRunner
   -> GcSchedule m
   -> Word64 -- ^ Number of immutable blocks replayed on ledger DB startup
   -> m Void
-copyAndSnapshotRunner cdb@CDB{..} gcSchedule replayed =
-    if onDiskShouldTakeSnapshot Nothing replayed then do
-      updateLedgerSnapshots cdb
-      now <- getMonotonicTime
-      loop (Just now) 0
-    else
-      loop Nothing replayed
+copyAndSnapshotRunner cdb@CDB{..} gcSchedule replayed = do
+    snapshotCandidate <- atomically $ LgrDB.oldestLedgerSnapshot <$> LgrDB.getCurrent cdbLgrDB
+    case getTipSlot snapshotCandidate of
+      Origin ->
+        loop Nothing replayed
+      NotOrigin slotNo ->
+        if onDiskShouldTakeSnapshot Nothing replayed slotNo then do
+          updateLedgerSnapshots cdbLgrDB snapshotCandidate
+          now <- getMonotonicTime
+          loop (Just now) 0
+        else
+          loop Nothing replayed
   where
     SecurityParam k      = configSecurityParam cdbTopLevelConfig
     LgrDB.DiskPolicy{..} = LgrDB.getDiskPolicy cdbLgrDB
 
+    -- getSlotNo :: ExtLedgerState blk -> WithOrigin SlotNo
```

Comment:
There is commented out code

Reply:
removed in the commit it was introduced (rebased)